### ISSUE TITLE
Dark Mode for Web Components

### ISSUE THREAD

**Issue by enowak1031** [trimble-oss/modus-web-components#739]
### Theme Setting

Implement 'dark' and 'light' theme setting for the web components
*Maybe* add a 'preference' mode which will automatically apply the user's preferred theme? This could be its own story outside of the Barebones requirements

There should be two ways to set the theme:
- [x] Page Level - Set the theme on the page's body with a class that the components can read from. This would apply the theme globally
- [x] Component Level - Set the theme directly on the component with a CSS attribute, which would override the Page Level setter

---

### 

- [x] #787
- [x] #788 
- [x] #789 
- [x] #796
- [x] #797
- [x] #800
- [x] #801
- [x] #802
- [x] #803
- [x] #804
- [x] #805
- [ ] #843
- [x] #853
- [x] #874
- [ ] maybe #847

Once we've got our pattern for setting the theme on the Page Level, a `ThemeToggle` component should be created that will switch the theme globally. This will allow consumers of the library to get 'out of the box' theme toggling
